reasoning/src/commonTest/kotlin/serpapi/SerpApiClientSpec.kt

```diff
@@ -0,0 +1,55 @@
+package serpapi
+
+import com.xebia.functional.tokenizer.ModelType
+import com.xebia.functional.xef.conversation.Conversation
+import com.xebia.functional.xef.store.ConversationId
+import com.xebia.functional.xef.store.LocalVectorStore
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.shouldBe
+import kotlinx.uuid.UUID
+import kotlinx.uuid.generateUUID
+import mocks.TestEmbeddings
+import mocks.TestModel
+
+class SerpApiClientSpec :
+  StringSpec({
+    "content should be the same" {
+      val conversationId = ConversationId(UUID.generateUUID().toString())
+
+      val model = TestModel(modelType = ModelType.ADA, name = "fake-model")
+
+      val scope = Conversation(LocalVectorStore(TestEmbeddings()), conversationId = conversationId)
+
+      val search = TestSearch(model = model, scope = scope)
+
+      val response = search.invoke("number of Leonardo di Caprio's girlfriends")
+
+      val contentExpected =
+        "Content: Gigi Hadid · Camila Morrone · Camila Morrone · Camila Morrone · Nina Agdal · Rihanna · Rihanna · Kelly Rohrbach."
```

Comment:
few names here are repeated; also, we should leave Leonardo in peace and use different examples without having to mention a specific person

Reply:
Done